src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -231,7 +233,8 @@ public void authorize(ClientState state) throws InvalidRequestException, Unautho
 
     public void validate(ClientState state) throws InvalidRequestException
     {
-        // Nothing to do, all validation has been done by RawStatement.prepare()
+        if (parameters.allowFiltering && !SchemaConstants.isSystemKeyspace(table.keyspace))
+            checkTrue(DatabaseDescriptor.getGuardrailsConfig().getAllowFilteringEnabled(), "ALLOW FILTERING statement is prohibited");
```

Comment:
This isn't verifying the `ClientState` nor the global enabling of guardrails. Instead of directly verifying the config property we should use the guardrail itself:
```suggestion
            Guardrails.allowFilteringEnabled.ensureEnabled(state);
```
There is a note about this [here](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/guardrails/GuardrailsConfig.java#L26-L28).

Unfortunately that makes the new tests fail, because `CQLTester` uses an internal query that is excluded from guardrails checks. Probably it would be easier to add a new test class extending `GuardrailTester`, which contains utilities for this. That class would be very similar to, for example, `GuardrailUserTimestampsTest`, which is also used to test another `DisabledFlag`s.

Reply:
thank you for the clarification, I was unaware of this but will update the function now.